app/services/slack/api_client.rb

```diff
@@ -2,20 +2,23 @@
 
 class Slack::ApiClient
   API_TOKEN = ENV.fetch("SLACK_API_TOKEN")
-  API_HOST = "https://slack.com/api".freeze
+  API_HOST = "https://slack.com".freeze
   private_constant :API_TOKEN
   private_constant :API_HOST
 
-  def initialize(network_provider=Faraday.new)
+  def self.build
+    new(Faraday.new(API_HOST))
+  end
+
+  def initialize(network_provider)
     @network_provider = network_provider
   end
 
   def get_user_data(slack_id)
-    response = @network_provider.get do |req|
-      req.url "#{API_HOST}/users.info"
-      req.params["token"] = API_TOKEN
-      req.params["user"] = slack_id
-    end
+    response = @network_provider.get(
+      "/api/users.info",
+       { user: slack_id }.merge(default_params)
```

Comment:
i think that reversing this would look better (default suggest it can be overriden) and allow easier change in future
`default_params.merge(user: slack_id)` or `{ user: slack_id }.reverse_merge(default_params)`